samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManagerFactory.scala

```diff
@@ -48,9 +49,24 @@ object KafkaCheckpointManagerFactory {
     } else {
       new KafkaConfig(config).getCheckpointSegmentBytes()
     }
-    (new Properties /: Map(
-      "cleanup.policy" -> "compact",
-      "segment.bytes" -> String.valueOf(segmentBytes))) { case (props, (k, v)) => props.put(k, v); props }
+
+    val isStreamMode = if (config != null) {
+      val appConfig = new ApplicationConfig(config)
+      appConfig.getAppMode == ApplicationMode.STREAM
+    } else {
+      true
+    }
+
+    if (isStreamMode) {
+      (new Properties /: Map(
+        "cleanup.policy" -> "compact",
+        "segment.bytes" -> String.valueOf(segmentBytes))) { case (props, (k, v)) => props.put(k, v); props }
+    } else {
+      (new Properties /: Map(
+        "cleanup.policy" -> "compact,delete",
```

Comment:
Can we consolidate this logic, and the one in `KafkaConfig` in one place?

Reply:
Good suggestion. I moved the logic there and now it looks much cleaner.